nussl/separation/duet.py

```diff
@@ -297,6 +356,63 @@ def find_peaks(data, min_thr=0.5, min_dist=None, max_num=1):
 
         return peak_indices
 
+    def compute_masks(self, atn_peak, delay_peak):
+        """Receives the attenuation and delay peaks and computes a mask to be applied to the signal for source
+        separation.
+
+        Parameters:
+            atn_peak (np.array): Attenuation peaks determined from histogram
+            delay_peak (np.array): Delay peaks determined from histogram
+        Returns:
+           best_ind(np.array): The indices for where the masks are applied
+           mask (np.array): A binary mask used to separate the sources
+
+        """
+        # compute masks for separation
+        best_so_far = np.inf * np.ones((self.num_frequency_bins - 1, self.num_time_bins))
+        best_ind = np.zeros((self.num_frequency_bins - 1, self.num_time_bins), dtype=int)
+        mask = None
```

Comment:
Right now this mask looks like this (in 1D): `[ 0, 0, 3, 2, 0, 2, 2, 1, 1, 3, 3 ]` where each number is a mask. This is not the most organized way to do this. I'm going to have to look at the new `MaskBase` methods and use those instead here. This method should return a list of `MaskBase`-derived objects instead of this Frankenstein mask array.